### PR TITLE
Refactor CVE fix into self-contained skill

### PR DESCRIPTION
**Pull request by dfarrell07**
Move the CVE fix automation from scripts/cve/ to
skills/cve-fix/ as a self-contained Claude Code skill
with CLI support via Makefile.

Key capabilities:
- Automated scan, fix, ignore, test, and PR generation
- Multi-repo mode (make all BRANCH=0.24)
- Deterministic fixes with LLM review for edge cases
- Subagent verification (yaml-integrity, fix-availability,
  scan-regression, existing-entries)
- Auto-expiring ignores via grype fix-state matching
- Transitive dependency detection and fixing
- Per-repo configuration via .cve-fix.yaml

Fixes discovered during production testing:
- Use pipe delimiter to prevent bash field collapsing
- Extract actual grype fix-state per CVE (GO-* vs GHSA-*)
- Handle transitive deps not listed in go.mod
- Group scan-regression CVEs per package for efficiency
- Clean stale ignores after fix-availability bumps
- Enforce 72-char body line limit in commit messages
- Clean build artifacts before final scan
- Use for-loop instead of xargs in make-all
- Lowercase worktree path for Docker compatibility

## Summary by CodeRabbit

* **New Features**
  * Added a centralized CVE remediation workflow for individual or multiple repositories.
  * Added configurable build, cleanup, testing, vendor, and environment settings.
  * Added automated dependency and Go standard-library updates, vulnerability scanning, ignore handling, and regression checks.
  * Added structured review reporting for unresolved vulnerabilities and configuration issues.

* **Documentation**
  * Added usage guidance, troubleshooting instructions, commit templates, and review procedures.

* **Chores**
  * Replaced the previous CVE command workflow with the new skill-based process.

### CODE REVIEW

**Review by vthapar — APPROVED** [review]

File: scripts/cve/review-prompt.md [hunk 1]
```diff
@@ -28,9 +28,11 @@ Review ALL CVE outcomes:
 
 2. **Handle unfixed CVEs**: For each CVE that was not fixed:
    - Try a different approach if possible (different version, drop replace directive)
-   - If fix would break the branch: add to .grype.yaml ignore list.
-     **Batch all CVEs for the same package into one ignore.sh call** (it accepts multiple CVE IDs).
+   - If **no fix is available**: use `--no-fix` so the entry auto-expires when a fix is published.
+     Run: `bash ${CVE_SCRIPTS}/ignore.sh ${STATE_FILE} PACKAGE SEVERITY "reason" --no-fix CVE_ID [CVE_ID...]`
+   - If **fix exists but would break the branch**: omit `--no-fix` (permanent ignore for this branch).
      Run: `bash ${CVE_SCRIPTS}/ignore.sh ${STATE_FILE} PACKAGE SEVERITY "reason" CVE_ID [CVE_ID...]`
+   - **Batch all CVEs for the same package into one ignore.sh call** (it accepts multiple CVE IDs).
```
[the remark this answers is not in the document]
Comment (dfarrell07) [reply scripts/cve/review-prompt.md]: Fixed, markdownlint passes clean.

File: scripts/cve/fix-stdlib.sh [hunk 2]
```diff
@@ -30,6 +30,11 @@ if [[ "$OLD_MINOR" != "$NEW_MINOR" ]]; then
   exit 2
 fi
 
+# For patch bumps, auto-download the needed Go version
+if [[ "$OLD_MINOR" == "$NEW_MINOR" ]] && [[ "${GOTOOLCHAIN:-}" != *auto* ]]; then
+  export GOTOOLCHAIN=auto
```
[the remark this answers is not in the document]
Comment (dfarrell07) [reply scripts/cve/fix-stdlib.sh]: Good catch — fixed. Now only sets `GOTOOLCHAIN=auto` when unset, preserving explicit caller values.

**Comment by dfarrell07** [comment]
@coderabbitai review

**Comment by dfarrell07** [comment]
@coderabbitai review

File: skills/cve-fix/Makefile [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+fix:
+	$(if $(REPO),,$(error REPO is required (name from repos.yaml or absolute path)))
+	$(if $(BRANCH),,$(error BRANCH is required (e.g., 0.23, release-0.23, or devel)))
+	./scripts/fix-all.sh "$(REPO)" "$(BRANCH)"
+
+all:
+	$(if $(BRANCH),,$(error BRANCH is required for all))
+	@for repo in $$(grep '^\s*- name:' repos.yaml | awk '{print $$NF}'); do \
+		$(MAKE) fix REPO=$$repo BRANCH=$(BRANCH) || true; \
+	done
```
[the remark this answers is not in the document]
Comment (dfarrell07) [reply skills/cve-fix/Makefile]: Fixed. The `all` target now propagates the worst exit code: `rc=0; ... || { err=$?; [ $err -gt $rc ] && rc=$err; }; exit $rc`. Returns exit 2 if any repo has unresolved CVEs, exit 1 for fatal errors, exit 0 only when all succeed.

File: skills/cve-fix/scripts/clean.sh [hunk 4]
```diff
@@ -1,5 +1,5 @@
 #!/bin/bash
 # Kill orphaned CVE fix processes and containers.
-# Safe to run anytime — only targets fix-all.sh, grype, and dapper fix containers.
+# Safe to run anytime — only targets fix-all.sh, grype, and build containers.
 pkill -f fix-all.sh 2>/dev/null || true
 docker ps --format '{{.ID}} {{.Image}}' 2>/dev/null | grep -E 'grype:latest|fix-.*-cves-' | awk '{print $1}' | xargs -r docker kill 2>/dev/null || true
```
[the remark this answers is not in the document]
Comment (dfarrell07) [reply skills/cve-fix/scripts/clean.sh:5]: Intentional. `clean.sh` is a manual emergency cleanup tool (`make clean`), not called automatically during `make all`. The `for`-loop runs repos sequentially so only one `fix-all.sh` process exists at a time. A user running `make clean` explicitly wants to kill all orphaned processes.

File: skills/cve-fix/scripts/fix-all.sh [hunk 5]
```diff
@@ -0,0 +1,309 @@
...
+  ] |
+  # Dedup: when same (pkg, cve) has both fix and no-fix entries, keep fix
+  group_by(.pkg + ":" + .cve) |
+  map(sort_by(if .type == "no-fix" then 1 else 0 end) | .[0]) |
+  group_by(.pkg + ":" + .type + ":" + (if .type == "no-fix" then .fixState else "" end)) |
+  map({
+    pkg: .[0].pkg,
+    type: .[0].type,
+    fixedIn: (if .[0].type == "no-fix" then ""
+              else (map(.fixedIn) | map(select(. != "")) | sort_by(split("-")[0] | split(".") | map(tonumber)) | last // "") end),
```
[the remark this answers is not in the document]
Comment (dfarrell07) [reply skills/cve-fix/scripts/fix-all.sh:110]: The `sort_by(split("-")[0] | split(".") | map(tonumber))` handles Go module versions which are always clean semver (no `v` prefix in grype's `fix.versions` JSON output). Verified across all 7 Submariner repos — zero v-prefixed fix versions. The `split("-")[0]` strips pre-release suffixes. If jq fails, the pipeline's `|| echo ""` prevents script crash.

File: skills/cve-fix/scripts/fix-all.sh [hunk 6]
```diff
@@ -0,0 +1,309 @@
...
+
+# --- Subagent Verification ---
+banner "Subagent Verification"
+SUBAGENT_DIR=""
+SUBAGENT_DIR="$(cd "$SCRIPT_DIR/../subagents" 2>/dev/null && pwd || true)"
+[[ -d "$SUBAGENT_DIR" ]] || SUBAGENT_DIR=""
+
+SUBAGENT_REPORT_DIR="/tmp/cve-fix-subagents-$(basename "$REPO")-${BRANCH//\//-}-$$"
+mkdir -p "$SUBAGENT_REPORT_DIR"
+printf '%s\n' "$SCAN_JSON" > "$SUBAGENT_REPORT_DIR/initial-scan.json"
```
[the remark this answers is not in the document]
Comment (dfarrell07) [reply skills/cve-fix/scripts/fix-all.sh:249]: The path includes `$$` (PID) for uniqueness. This is a local developer tool, not a service — the threat model of symlink attacks doesn't apply. The directory contains only CVE scan JSON (non-sensitive). `mktemp -d` would require refactoring the cleanup trap which references the path by convention.

File: skills/cve-fix/scripts/lib.sh [hunk 7]
```diff
@@ -0,0 +1,584 @@
...
+      evidence=$(bash "$sh_file" "$state_file" 2>&1) || true
+    fi
+    printf '%s\n' "$evidence" > "$report_dir/${sa_name}.evidence"
+  done
+
+  # Phase 2: Launch subagents in parallel
+  if ! command -v claude &>/dev/null; then
+    echo "  WARNING: claude CLI not available, skipping subagent verification."
+    return 0
+  fi
```
[the remark this answers is not in the document]
Comment (dfarrell07) [reply skills/cve-fix/scripts/lib.sh:286]: Unreachable code path. The caller at `fix-all.sh:251` already guards with `command -v claude` before calling `verify_fix_loop`. The `return 0` inside `run_subagents` is defense-in-depth. Companion scripts produce `.evidence` files but cannot produce verdicts without the LLM subagents — failing here would be counterproductive.

File: skills/cve-fix/subagents/existing-entries.sh [hunk 8]
```diff
@@ -0,0 +1,132 @@
...
+  fi
+
+  # Also check tools/go.mod
+  if [[ "$HAS_TOOLS_GOMOD" == "true" ]] && [[ -z "$CURRENT_VER" ]]; then
+    CURRENT_VER=$(go -C tools list -m "$PKG" 2>/dev/null | awk '{print $2}' || echo "")
+    VERSIONS=$(go -C tools list -m -versions "$PKG" 2>/dev/null | sed "s|^$PKG ||" || echo "")
+    if [[ -n "$VERSIONS" ]]; then
+      LATEST_VER=$(echo "$VERSIONS" | tr ' ' '\n' | tail -1)
+    fi
+  fi
```
[the remark this answers is not in the document]
Comment (dfarrell07) [reply skills/cve-fix/subagents/existing-entries.sh:104]: False positive. `HAS_TOOLS_GOMOD` IS written to the state file at `detect.sh:195`. The variable is always set to `true` or `false` before the state file is written. Verified: `grep HAS_TOOLS_GOMOD skills/cve-fix/scripts/detect.sh` shows it on line 67 (detection) and line 195 (state file output).

File: skills/cve-fix/subagents/fix-availability.md [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+- A newer version of the package exists in the Go module proxy
+- This strongly suggests a fix is available and the `fix-state: not-fixed` entry
+  may be stale or was added prematurely
+- The package should be fixed rather than ignored
+
+## Verdict
+
+- **PASS** if `NEW_ISSUES=0` -- all not-fixed/unknown entries are genuine (no newer versions available)
+- **FAIL** if any `FIX-AVAILABLE` entries exist -- a fix may be available for packages
+  currently marked as unfixable
```
[the remark this answers is not in the document]
Comment (dfarrell07) [reply skills/cve-fix/subagents/fix-availability.md:26]: The wording difference (`not-fixed` in verdict vs `not-fixed/unknown` in task description) is cosmetic. The companion script correctly checks both values (`=~ ^(not-fixed|unknown)$`). The LLM subagent sees the companion evidence which includes the actual fix-state values.

File: skills/cve-fix/subagents/scan-regression.sh [hunk 10]
```diff
@@ -0,0 +1,93 @@
...
+else
+  NEW_CVE_LIST=""
+fi
+
+while IFS= read -r cve; do
+  [[ -z "$cve" ]] && continue
+  # Get package info from final scan (include fix version for deterministic fix)
+  PKG_INFO=$(echo "$FINAL_SCAN" | jq -r --arg cve "$cve" \
+    '.matches[] | select(.vulnerability.id == $cve) | "\(.artifact.name) \(.artifact.version) \(.vulnerability.severity) \(((.vulnerability.fix.versions // []) | if length == 0 then "no-fix" else (sort_by(split("-")[0] | split(".") | map(tonumber)) | last) end))"' \
+    2>/dev/null | head -1 || echo "unknown")
```
[the remark this answers is not in the document]
Comment (dfarrell07) [reply skills/cve-fix/subagents/scan-regression.sh:63]: Same as the fix-all.sh:110 comment — grype's Go module `fix.versions` are always bare semver without `v` prefix. Verified with live grype scans across all 7 repos. The `2>/dev/null` on the jq pipeline handles any edge cases gracefully, and the consumer (`subagent_fix_scan_regression`) strips `v` prefix defensively at line 542.

File: skills/cve-fix/subagents/yaml-integrity.sh [hunk 11]
```diff
@@ -0,0 +1,96 @@
...
+# Check fix-state values
+FIX_STATES=$(echo "$ADDED_LINES" | grep 'fix-state:' | sed 's/.*fix-state:[[:space:]]*//' || true)
+
+while IFS= read -r fix_state; do
+  [[ -z "$fix_state" ]] && continue
+  TRIMMED=$(echo "$fix_state" | xargs)
+  if [[ "$TRIMMED" != "not-fixed" && "$TRIMMED" != "unknown" && "$TRIMMED" != "wont-fix" ]]; then
+    ISSUES+=("BAD_FIX_STATE: fix-state '$TRIMMED' is not 'not-fixed', 'unknown', or 'wont-fix'")
+  fi
+done <<< "$FIX_STATES"
```
[the remark this answers is not in the document]
Comment (dfarrell07) [reply skills/cve-fix/subagents/yaml-integrity.sh:70]: Global validation is equivalent to per-entry validation for field-value checks. All entries are generated by `insert_grype_ignore` which always produces complete entries (comment + vulnerability + optional fix-state + package name). A manual edit creating an incomplete entry is caught by the comment-line check at line 77-88. The theoretical scenario of a name-less entry passing due to another entry having a name cannot occur through the automated pipeline.

File: skills/cve-fix/subagents/yaml-integrity.sh [hunk 12]
```diff
@@ -0,0 +1,96 @@
...
+    if ! echo "$PREV_LINE" | grep -q '^\s*#'; then
+      VULN=$(echo "$line" | sed 's/.*vulnerability:[[:space:]]*//')
+      # Only flag if this is a newly added entry
+      if echo "$ADDED_LINES" | grep -qF "vulnerability: $VULN"; then
+        ISSUES+=("MISSING_COMMENT: no comment line before entry for $VULN")
+      fi
+    fi
+  fi
+  PREV_LINE="$line"
+done <<< "$FULL_GRYPE"
```
[the remark this answers is not in the document]
Comment (dfarrell07) [reply skills/cve-fix/subagents/yaml-integrity.sh:88]: The commit-then-verify flow makes `HEAD` correct. `ignore.sh` always commits before yaml-integrity runs (`verify_fix_loop` in lib.sh calls `run_subagents` after all fix functions have committed). The `|| cat .grype.yaml` fallback handles the edge case where HEAD doesn't have the file yet. Reading the working tree instead would be wrong if uncommitted changes exist from a failed fix attempt.

File: .agents/commit-templates.md [hunk 13]
```diff
@@ -56,6 +56,35 @@ Full package: helm.sh/helm/v3
...
+
+<reason>
+```
+
+**Fix exists but incompatible with this branch:**
+```text
+Ignore <abbreviated-package> CVEs incompatible with <branch>
+
+<reason>
+```
```
[the remark this answers is not in the document]
Comment (dfarrell07) [reply .agents/commit-templates.md:75]: Documentation covers the most common abbreviation patterns. The `abbreviate_package` function in lib.sh handles all cases programmatically — the templates are examples, not an exhaustive list.

File: skills/cve-fix/scripts/detect.sh [hunk 14]
```diff
@@ -13,20 +13,20 @@ source "$SCRIPT_DIR/lib.sh"
...
-        arg="../$arg"
-      fi
-      if [[ "$arg" == /* ]] || [[ "$arg" == ./* ]] || [[ "$arg" == ../* ]] || [[ -d "$arg" ]]; then
+      if [[ "$arg" == /* ]]; then
         [[ -n "$REPO" ]] && { echo "ERROR: Multiple repositories specified" >&2; exit 1; }
         REPO="$arg"
+      elif [[ -f "$REPOS_YAML" ]] && grep -q "^  - name: ${arg}$" "$REPOS_YAML"; then
+        [[ -n "$REPO" ]] && { echo "ERROR: Multiple repositories specified" >&2; exit 1; }
+        REPO=$(grep -A1 "^  - name: ${arg}$" "$REPOS_YAML" | grep 'path:' | awk '{print $2}')
+        REPO="${REPO/#\~/$HOME}"
```
[the remark this answers is not in the document]
Comment (dfarrell07) [reply skills/cve-fix/scripts/detect.sh:29]: The repos.yaml schema is project-controlled with a fixed format. The grep/awk parsing matches the exact YAML structure. Adding a YAML parser would add a dependency for no practical benefit.

File: skills/cve-fix/scripts/fix-package.sh [hunk 15]
```diff
@@ -0,0 +1,188 @@
...
+      BODY="Full package: $PACKAGE
+Fixes:
+$FIXES"
+    else
+      BODY="Full package: $PACKAGE
+$FIXES_LINE"
+    fi
+  fi
+
+  git commit -s -m "$(printf '%s\n\n%s' "$SUBJECT" "$BODY")"
```
[the remark this answers is not in the document]
Comment (dfarrell07) [reply skills/cve-fix/scripts/fix-package.sh:183]: The 50-char limit has a fallback (multi-CVE format). Extremely long abbreviated package names (e.g., Azure SDK deep paths) can still exceed 50 in the fallback, but hard truncation would garble the subject. The 50-char convention is a guideline, not enforced by git.

File: skills/cve-fix/scripts/fix-stdlib.sh [hunk 16]
```diff
@@ -0,0 +1,120 @@
...
+# shellcheck source-path=SCRIPTDIR
+# shellcheck source=lib.sh
+source "$SCRIPT_DIR/lib.sh"
+
+STATE_FILE="${1:?Usage: fix-stdlib.sh STATE_FILE GO_VERSION ALL_FIX_VERSIONS CVE_ID [CVE_ID...]}"
+GO_VERSION="${2:?Missing GO_VERSION}"
+ALL_FIX_VERSIONS="${3:-}"
+shift 3
+CVE_IDS=("$@")
+[[ ${#CVE_IDS[@]} -gt 0 ]] || { echo "ERROR: At least one CVE_ID required" >&2; exit 1; }
```
[the remark this answers is not in the document]
Comment (dfarrell07) [reply skills/cve-fix/scripts/fix-stdlib.sh:17]: All callers (fix-all.sh) always provide all 3 positional args before CVE IDs. Missing args would also fail at the CVE_IDS check on line 16.

File: skills/cve-fix/scripts/fix-stdlib.sh [hunk 17]
```diff
@@ -0,0 +1,120 @@
...
+# Check host Go version is sufficient
+HOST_GO=$(go version | grep -oP 'go\K[0-9]+\.[0-9]+\.[0-9]+')
+if [[ -n "$HOST_GO" ]] && [[ "${GOTOOLCHAIN:-}" != *auto* ]] && \
+   [[ "$(printf '%s\n' "$HOST_GO" "$GO_VERSION" | sort -V | head -1)" == "$HOST_GO" ]] && \
+   [[ "$HOST_GO" != "$GO_VERSION" ]]; then
+  echo "NEEDS_REVIEW: stdlib — host Go $HOST_GO is older than required $GO_VERSION"
+  echo "Update Go: go install golang.org/dl/go${GO_VERSION}@latest && go${GO_VERSION} download"
+  echo "Or set GOTOOLCHAIN=auto to allow automatic toolchain downloads."
+  exit 2
+fi
```
[the remark this answers is not in the document]
Comment (dfarrell07) [reply skills/cve-fix/scripts/fix-stdlib.sh:78]: The script requires `go` for `go mod edit` and `go mod tidy` at lines 83-85. Failing at the version check vs failing at mod edit produces the same outcome under `set -e`.

File: skills/cve-fix/scripts/ignore.sh [hunk 18]
```diff
@@ -0,0 +1,82 @@
...
+fi
+
+# Enforce 50-char subject limit (full detail is always in the body)
+if [[ ${#SUBJECT} -gt 50 ]]; then
+  if [[ -n "$FIX_STATE" ]]; then
+    SUBJECT="Ignore $ABBREV CVEs (no fix)"
+  else
+    SUBJECT="Ignore $ABBREV CVEs ($BRANCH)"
+  fi
+fi
```
[the remark this answers is not in the document]
Comment (dfarrell07) [reply skills/cve-fix/scripts/ignore.sh:73]: Same as fix-package.sh:183. The short-form fallback (`(no fix)` / `($BRANCH)`) handles most cases. Hard truncation would lose meaning. A few chars over 50 is acceptable for the convention.

File: skills/cve-fix/scripts/lib.sh [hunk 19]
```diff
@@ -0,0 +1,584 @@
+# shellcheck shell=bash
+# Shared functions for CVE fix scripts
+
+# Compute state file path from repo path and branch name
+state_file_path() {
+  local repo_basename branch_sanitized
+  repo_basename=$(basename "$1")
+  branch_sanitized=$(echo "$2" | tr './' '-')
+  echo "/tmp/cve-fix-${repo_basename}-${branch_sanitized}-$$.env"
+}
```
[the remark this answers is not in the document]
Comment (dfarrell07) [reply skills/cve-fix/scripts/lib.sh:10]: Same as fix-all.sh:249. Local developer tool with PID-based uniqueness. The state file contains repo paths and branch names, not secrets.

File: skills/cve-fix/scripts/test-lib.sh [hunk 20]
```diff
@@ -0,0 +1,634 @@
...
+
+# Should have 3 calls: fix for mixed, ignore for mixed, fix for other
+MIX_CALL_COUNT=$(wc -l < "$MOCK_LOG_MIX" 2>/dev/null || echo 0)
+check "mixed: 3 calls" test "$MIX_CALL_COUNT" -eq 3
+check "mixed: fix call" bash -c "grep 'MOCK_FIX.*mixed' '$MOCK_LOG_MIX' | grep -q 'GHSA-1234-5678-abcd'"
+check "mixed: ignore call" bash -c "grep 'MOCK_IGNORE.*mixed' '$MOCK_LOG_MIX' | grep -q 'GHSA-abcd-1234-5678'"
+check "mixed: other fix" bash -c "grep 'MOCK_FIX.*other' '$MOCK_LOG_MIX' | grep -q 'CVE-2026-9999'"
+
+# Verify fix_state: GHSA- prefix should trigger not-fixed
+check "mixed: GHSA gets not-fixed" bash -c "grep 'MOCK_IGNORE.*mixed' '$MOCK_LOG_MIX' | grep -q 'GHSA-abcd-1234-5678'"
```
[the remark this answers is not in the document]
Comment (dfarrell07) [reply skills/cve-fix/scripts/test-lib.sh]: Fixed. The mock now logs `FIX_STATE=$FIX_STATE` and the assertion verifies `FIX_STATE=not-fixed` instead of just re-checking the package name.

File: skills/cve-fix/subagents/fix-availability.md (hunk 9 above)
[the remark this answers is not in the document]
Comment (dfarrell07) [reply skills/cve-fix/subagents/fix-availability.md:26]: The heuristic-then-verify architecture is intentional. fix-availability is a trigger, not a verdict. The chain is: fix-availability detects newer version → subagent_fix_fix_availability bumps it → next verify_fix_loop iteration runs scan-regression with a fresh grype scan → if the CVE persists, scan-regression catches it as UNRESOLVED and the fix function handles it. CVE-specific verification happens at the scan-regression layer, not the fix-availability layer.

File: skills/cve-fix/subagents/yaml-integrity.sh (hunk 11 above)
[the remark this answers is not in the document]
Comment (dfarrell07) [reply skills/cve-fix/subagents/yaml-integrity.sh:70]: Agreed that per-entry structural validation would be defense-in-depth, but it addresses no real failure mode. insert_grype_ignore is the sole write path and unconditionally produces complete entries. The review prompt forbids direct edits, and subagent_fix_yaml_integrity resets and re-applies via ignore.sh on any formatting failure. The script is a smoke test for field values, not a structural YAML parser — structural completeness is guaranteed architecturally.

File: skills/cve-fix/scripts/lib.sh (hunk 19 above)
[the remark this answers is not in the document]
Comment (dfarrell07) [reply skills/cve-fix/scripts/lib.sh:10]: Explicitly accepted for this developer-only tool. The risk is mitigated by: (1) fs.protected_symlinks=1 (default on all major Linux distros since ~2012) blocks symlink following in /tmp, (2) PID suffix makes path unpredictable, (3) state file contains only non-sensitive build config (paths, branch names, boolean flags), (4) no multi-tenant deployment scenario. Using mktemp would require a different path-communication mechanism between scripts for no practical security benefit.

File: .agents/commit-templates.md (hunk 13 above)
[the remark this answers is not in the document]
Comment (dfarrell07) [reply .agents/commit-templates.md:75]: The templates show the full-form subjects. The scripts automatically fall back to shorter forms when the subject exceeds 50 chars: `(no fix)` instead of `(no fix available)`, `($BRANCH)` instead of `incompatible with $BRANCH`. These are implementation details of the length enforcement in ignore.sh (lines 60-67), not separate template patterns — documenting them as alternatives would imply a choice that doesn't exist (the script picks the form based on length automatically).

**Comment by dfarrell07** [comment]
@coderabbitai review

File: skills/cve-fix/SKILL.md [hunk 21]
```diff
@@ -41,35 +43,58 @@ the PR command.
...
+   `bash SKILL_DIR/scripts/fix-all.sh NAME BRANCH` via the Bash
+   tool (not the Skill tool, which times out in subagents).
 
-For multiple repos, spawn one agent per repo. Each agent should run
-`bash ~/go/src/submariner-io/shipyard/scripts/cve/fix-all.sh REPO BRANCH`
-via the Bash tool (not the Skill tool, which times out in subagents).
 Report per repo: CVEs found, fixed, ignored, and PR command. On errors
-or timeout, clean up orphaned processes with
-`bash ~/go/src/submariner-io/shipyard/scripts/cve/clean.sh` before reporting.
+or timeout, run `bash SKILL_DIR/scripts/clean.sh` before reporting.
```
[the remark this answers is not in the document]
Comment (dfarrell07) [reply skills/cve-fix/SKILL.md:60]: The SKILL.md multi-repo section tells the Claude agent to run clean.sh on errors. In practice, the Agent tool runs repos sequentially (one completes before the next starts), not truly in parallel. The clean.sh call is for the single failed repo's orphaned processes. Even if agents ran concurrently, each fix-all.sh uses its own process group (`kill -- -$$`) for cleanup — clean.sh is a last-resort manual tool, not the primary cleanup mechanism.

File: skills/cve-fix/Makefile [hunk 22]
```diff
@@ -0,0 +1,39 @@
...
+	$(if $(BRANCH),,$(error BRANCH is required (e.g., 0.23, release-0.23, or devel)))
+	./scripts/fix-all.sh "$(REPO)" "$(BRANCH)"
+
+all:
+	$(if $(BRANCH),,$(error BRANCH is required for all))
+	@rc=0; \
+	for repo in $$(grep '^\s*- name:' repos.yaml | awk '{print $$NF}'); do \
+		$(MAKE) fix REPO=$$repo BRANCH=$(BRANCH) || { err=$$?; [ $$err -gt $$rc ] && rc=$$err; }; \
+	done; \
+	exit $$rc
```
[the remark this answers is not in the document]
Comment (dfarrell07) [reply skills/cve-fix/Makefile:31]: REPO and BRANCH are developer-supplied arguments on a local CLI tool, not untrusted user input. The values come from the developer typing `make fix REPO=admiral BRANCH=0.24` or from repos.yaml (project-controlled). Quoting `\"$(REPO)\"` on line 23 already wraps the value. Shell injection would require the developer to deliberately inject quotes into their own make arguments, which is self-attack. Adding input validation would reject legitimate paths containing spaces or special characters.

File: skills/cve-fix/scripts/test-lib.sh [hunk 23]
```diff
@@ -0,0 +1,634 @@
...
+check "rm-exc: exclude kept" grep -q "^exclude:" "$TMPD/rm-exc.yaml"
+check "rm-exc: original kept" grep -q CVE-2015-5237 "$TMPD/rm-exc.yaml"
+
+# Insert then remove round-trip leaves file equivalent
+cp "$REPO_ROOT/.grype.yaml" "$TMPD/rm-roundtrip.yaml"
+cp "$TMPD/rm-roundtrip.yaml" "$TMPD/rm-roundtrip-orig.yaml"
+insert_grype_ignore "$TMPD/rm-roundtrip.yaml" GHSA-rt-0001 example.com/rt "Temp" "not-fixed"
+remove_grype_ignore "$TMPD/rm-roundtrip.yaml" GHSA-rt-0001 example.com/rt
+check "rm-roundtrip: no GHSA remains" ! grep -q GHSA-rt-0001 "$TMPD/rm-roundtrip.yaml"
+
```
[the remark this answers is not in the document]
Comment (dfarrell07) [reply skills/cve-fix/scripts/test-lib.sh:525]: Good suggestion — added `diff -q` assertion to verify the file is fully restored after insert+remove round-trip. 121/121 tests pass.

**Comment by dfarrell07** [comment]
@coderabbitai review

**Comment by dfarrell07** [comment]
@coderabbitai review

**Comment by dfarrell07** [comment]
@coderabbitai review

File: skills/cve-fix/scripts/test-lib.sh [hunk 24]
```diff
@@ -0,0 +1,635 @@
...
+# === clean_gomod ===
+echo "clean_gomod"
+printf 'module t\ngo 1.25.0\ntoolchain go1.25.1\n' > "$TMPD/go.mod"
+(cd "$TMPD" && clean_gomod)
+check "toolchain removed" test -z "$(grep toolchain "$TMPD/go.mod")"
+check "go kept" grep -q "^go 1.25.0" "$TMPD/go.mod"
+
+# === load_state ===
+echo "load_state"
+check "missing fails" ! load_state /tmp/nonexistent-cve-test.env
```
[the remark this answers is not in the document]
Comment (dfarrell07) [reply skills/cve-fix/scripts/test-lib.sh]: Fixed. The missing-state test now uses `"$TMPD/nonexistent-cve-test.env"` (under the test's fresh `mktemp -d`), guaranteeing absence and removing the shared `/tmp` collision risk. 121/121 tests pass.

**Comment by dfarrell07** [comment]
@coderabbitai review

**Comment by dfarrell07** [comment]
@coderabbitai review

**Review by skitt — APPROVED** [review]
Next time, to make reviewing easier, please move files around in one commit (possibly adjusting as required for the path changes), and change them in a separate commit.
